designs/sam_local_inlinecode.rst

```diff
@@ -0,0 +1,80 @@
+.. contents:: **Table of Contents**
+   :depth: 2
+   :local:
+
+``InlineCode`` debug
+=====================
+This is the design for a feature to provide local testing capabilities to functions defined via **InlineCode** key schema.
+
+What is the problem?
+--------------------
+When new functions are started an absolute path of the folder/file containing AWS Lambda code package need to be passed
+to docker deamon in order to bind-mount the path inside the container running the AWs Lambda runtime.
+
+As of today you can only test locally your AWS Lambda Functions if defined via ``CodeUri`` key and not ``InlineCode``.
+
+What will be changed?
+---------------------
+In this proposal, we will change the code in a way that, if the tested AWS Lambda function is codified with
+``InlineCode`` key, we will create a temporary local file based on the ``InlineCode`` value, passing the temp
+file to bind-mount inside the container and make it possible to locally test ``InlineCode`` authoer Functions.
+
+Success criteria for the change
+-------------------------------
+#. Support all programming languages supported by ``InlineCode``:
+   * Nodeds
+   * Python
+
+#. Local testing should just work for Functions authored with ``InlineCode``
+
+Out-of-Scope
+------------
+
+#. Anything not specifically related to ``InlineCode`` and sam local commands
+#. Provide support for Functions authored with ``InlineCode`` and leveraging ``!Sub`` pseudo function;
+   this can be object for future integration
+
+User Experience Walkthrough
+---------------------------
+
+
+Implementation
+==============
+
+Design
+------
+*Explain how this feature will be implemented. Highlight the components of your implementation, relationships*
+*between components, constraints, etc.*
+
+Design should be pretty simple. Write a tempfile and passing it to the Doker daemon as a bind-mount.
+
+
+Security
+--------
+
+*Tip: How does this change impact security? Answer the following questions to help answer this question better:*
+
+**What new dependencies (libraries/cli) does this change require?**
+TBD, most likely no external dependencies
+
+**Are you reading/writing to a temporary folder? If so, what is this used for and when do you clean up?**
+Yes, I'm writing the ``InlineCode`` function string to a temporary file (system tempdir)
```

Comment:
Are we going to restrict the file permissions at all for this tmp file and if so to what?

Reply:
I'm not really sure if makes any sense/bring any value.. I would prefer to have a suggestion on this point rather than stating anything about it